V0.7.1

BDonnot opened the pull request:
Changelog:
- [FIXED] a bug in the chronics making it not start at the appropriate time step
- [FIXED] a bug in "OneChangeThenNothing" agent that prevent it to be restarted properly.
- [FIXED] a bug with the generated docker file that does not update to the last version of the package.
- [FIXED] numpy, by default does not use the same datatype depending on the platform. We   ensure that floating value are always `np.float32` and integers are always `np.int32`
- [ADDED] a method to extract only some part of a chronic.
- [ADDED] a method to "fast forward" the chronics
- [UPDATED] test data and datasets are no longer included in the package distribution
- [UPDATED] a new function `make_with_download` that will make obsolete the "grid2op.download" script in future versions
- [UPDATED] the python "requests" package is now a dependency
- [ADDED] class `grid2op.Reward.CombinedScaledReward`: A reward combiner with linear interpolation to stay within a given range.
- [ADDED] `grid2op.Reward.BaseReward.set_range`: All rewards have a default setter for their `reward_min` and `reward_max` attributes.
- [ADDED] `grid2op.PlotGrid`: Revamped plotting capabilities while keeping the interface we know from `grid2op.Plot`
- [ADDED] `grid2op.replay` binary: This binary is installed with grid2op and allows to replay a runner log with visualization and gif export

Tezirg committed:
New grid2op.Reward: CombinedScaled

Adds a set_range function in BaseReward & create a new CombinedScaledReward

Tezirg committed:
Changelog

BDonnot committed:
adding utility to extract only part of the chronics

BDonnot committed:
fixing a bug on chronics with the initial datetime reading

BDonnot committed:
make the new make happen

BDonnot committed:
adding documentation for make_new and MakeEnv module

BDonnot committed:
adding some user utilities to inspect what is downloaded, where it is etc.

BDonnot committed:
fixing stuff for userutils

BDonnot committed:
work on some backward compatibility and docker scrtip

BDonnot committed:
fixing again the docker file

BDonnot committed:
adding other docker images

BDonnot committed:
adding back the chronics in the github

BDonnot committed:
fixing grid name file in the Settings*

BDonnot committed:
fix some other path because tests still don't pass on docker

BDonnot committed:
trying to fix the docker bug for new version

BDonnot committed:
last try for the day

BDonnot committed:
adding new template docker file and improving release script

Tezirg committed:
grid2op.Reward

Update some custom rewards. As for range computation can be computed over the interval [max;min] directly using linear interpolation. Adds a new reward LinesReconnectedReward

Tezirg committed:
Merge confilcts resolution from zrg-dev into v0.7.1

Tezirg committed:
Circle ci install edit mode

Tezirg committed:
GameplayReward

Respect advertised range

Tezirg committed:
LinesReconnected positive

Tezirg committed:
New submodule grid2op.PlotGrid & grid2op.Episode

Updated plots in new submodule and moves Episode related file to submodule.

Tezirg committed:
PlotGrid fix

Make sure env.render works. Improves imports

Tezirg committed:
PlotGird.PlotPltly

Tezirg committed:
Minor comments

Tezirg committed:
Circle ci

Remove pip install in edit mode as the test docker should already do it

Tezirg committed:
Fix download script import

Tezirg committed:
Moves download.py into approriate folder

Tezirg committed:
grid2op._utils cleanup

Moved utils functions into Space.space_utils as these are scoped to serialization (aka. Space)

BDonnot committed:
fixed numpy dtyp

Tezirg committed:
Changelog

Tezirg committed:
Fixing issue #59

There was a duplicate private member in Environment. env.time_remaining_before_reconnection and env.time_remaining_before_line_reconnection. The env.step was updating the later while the ruleschecker was testing the former. Kept the later and removes the formar attribute to prevent future mistakes

BDonnot committed:
adding the fast forward for chronics, finishing the change of numpy dtype

BDonnot committed:
modifying changelog

BDonnot committed:
merging with terzig

Tezirg commented on CHANGELOG.rst:
```diff
@@ -22,6 +22,24 @@ Change Log
+- [ADDED] a method to extract only some part of a chronic.
+- [ADDED] a method to "fast forward" the chronics
+- [UPDATED] test data and datasets are no longer included in the package distribution
+- [UPDATED] a new function `make_with_download` that will make obsolete the "grid2op.download" script in future versions
```
C'est `make_new` ca non ? Pas besoin de deprecate download  ?

Tezirg commented on Dockerfile:
```diff
@@ -24,19 +24,23 @@ RUN apt-get update && \
 
 # Retrieve Grid2Op
-RUN git clone https://github.com/rte-france/Grid2Op
+RUN git clone https://github.com/bdonnot/Grid2Op
```
Restore to `https://github.com/rte-france/Grid2Op` for the main docker container

Tezirg commented on grid2op/Action/BaseAction.py:
```diff
@@ -12,6 +12,7 @@
 
 import pdb
 
+from grid2op._utils import dt_int, dt_bool, dt_float
```
Je propose de mettre les typedef dans `grid2op.dtypes.py` puisque melanger des defines de types et la classe `FakePBar` me semble un peu brouillon

Tezirg commented on grid2op/Environment/BaseEnv.py:
```diff
@@ -1087,9 +1088,9 @@ def _reset_vectors_and_timings(self):
-        self.time_remaining_before_reconnection = np.full(shape=(self.n_line,), fill_value=0, dtype=np.int)
+        self.time_next_maintenance = np.zeros(shape=(self.n_line,), dtype=dt_int) - 1
+        self.duration_next_maintenance = np.zeros(shape=(self.n_line,), dtype=dt_int)
+        self.time_remaining_before_reconnection = np.full(shape=(self.n_line,), fill_value=0, dtype=dt_int)
```
time_remaining_before_reconnection -> time_remaining_before_line_reconnection

Tezirg commented on grid2op/MakeEnv/PathUtils.py:
```diff
@@ -0,0 +1,36 @@
+        try:
+            os.mkdir(data_path)
+        except:
+            raise RuntimeError("Impossible to create a directory in \"{}\". Make you can write there. If you don't "
```
`Make you can write there` -> Nonsense 
Je ne comprend pas l'interet du config file dans le home si le `DEFAULT_PATH_DATA` est aussi dans le home

Tezirg commented on utils/templateDockerFile_test:
```diff
@@ -0,0 +1,48 @@
+RUN git pull
+RUN git remote update
+RUN git fetch --all --tags
+RUN git checkout "tags/__VERSION__" -b "__VERSION__-branch"
```
Pas besoin de checkout le dernier tag pour le container test.

Tezirg commented on utils/templateDockerFile_test:
```diff
@@ -0,0 +1,48 @@
+    ca-certificates
+
+# Retrieve Grid2Op
+RUN git clone https://github.com/rte-france/Grid2Op
```
Le container test doit pointer vers BDonnot

Tezirg commented on utils/templateDockerFile_test:
```diff
@@ -0,0 +1,48 @@
+# Install Dependencies
+RUN pip3 install -U -e .[test]
+
+# force the "install" of test data in proper repo
```
Obligatoire ? Ajouter le -e devrait etre suffisant ?

BDonnot committed:
adding a license information about all the dependenciesé

Tezirg committed:
Remove duplicate code that came back due to merge error

BDonnot committed:
integrating review comments

BDonnot committed:
fixing weird conflict on test_Observation, improving test_Rules

BDonnot committed:
updating dockerfile to NOT use cache when building

Tezirg committed:
Space related grid2op._utils are now under grid2op.Space.space_utils

Tezirg committed:
templateDockerfiles -U flag is error prone

Docker doesn't support users, it runs as root. So no need for pip install -U flag.

Tezirg committed:
templateDockerfiles

Removes comment on git clone line

Tezirg committed:
templateDockerfile_test Expose port 80

Tezirg commented on grid2op/tests/test_Rules.py:
```diff
@@ -171,15 +171,15 @@ def test_PreventReconection(self):
 
         self.helper_action.legal_action = RulesChecker(legalActClass=PreventReconnection).legal_action
-        self.env.time_remaining_before_line_reconnection = np.full(shape=(self.env.backend.n_line,),
+        self.env.times_before_line_status_actionable[:] = np.full(shape=(self.env.backend.n_line,),
```
test_PreventReconection:
times_before_line_status_actionable or time_remaining_before_line_reconnection ?

Tezirg committed:
LineReconnected reward range fix

Tezirg committed:
PlotGrid make plotly export only if dep is installed

Tezirg committed:
Optional dependencies silent export fail

Tezirg committed:
grid2op.replay bianary wrning format

Tezirg committed:
Silent fail for optional dependencies only on importerror

Tezirg committed:
grid2op.Environment.render fail on import only

BDonnot committed:
adding license files

BDonnot committed:
fix a bug in the checking order of makenew

BDonnot committed:
Merge branch 'v0.7.1' of https://github.com/BDonnot/grid2op into v0.7.1

BDonnot committed:
last fix before release of v.0.7.1
